### PR TITLE
Mes 4434 debrief page cat a mod2

### PR DESCRIPTION
**Pull request by appleabs**
## Description
Mes-4434 Created the debrief page for category A Mod 2.
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/debrief/cat-a-mod2/__tests__/debrief.cat-a-mod2.page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    welshTest: false,
+    extendedTest: false,
+    slotId: 123,
+    specialNeeds: false,
+    start: '',
+    vehicleTypeCode: '',
+  };
+
+  beforeEach(async(() => {
+    const exampleTestData: CatBEUniqueTypes.TestData  = {
```
Comment (rossfauldsbjss) [inline src/pages/debrief/cat-a-mod2/__tests__/debrief.cat-a-mod2.page.spec.ts:66]: TODO to replace CatBE

File: src/pages/debrief/cat-a-mod2/components/vehicle-checks-card/__tests__/vehicle-checks-card.spec.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      providers: [
+        { provide: Config, useFactory: () => ConfigMock.instance() },
+      ],
+    })
+      .compileComponents()
+      .then(() => {
+        fixture = TestBed.createComponent(VehicleChecksCardCatAMod2Component);
+        store$ = TestBed.get(Store);
+
+        store$.dispatch(new StartTest(105, TestCategory.BE));
```
Comment (rossfauldsbjss) [inline src/pages/debrief/cat-a-mod2/components/vehicle-checks-card/__tests__/vehicle-checks-card.spec.ts]: TODOS for BE

File: src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  ionViewDidLeave(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+
+    }
+  }
+
+  endDebrief(): void {
+    this.store$.dispatch(new EndDebrief());
+    if (this.outcome === 'Pass') {
```
Comment (rossfauldsbjss) [inline src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts:179]: magic string - is there a definition anywhere of pass and fail?

**Review by rossfauldsbjss — APPROVED** [review]
looks good

File: src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts (hunk 3 above)
Comment (appleabs) [reply src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts]: I don't think so I can't seem to find one if there is

File: src/pages/debrief/cat-a-mod2/components/vehicle-checks-card/__tests__/vehicle-checks-card.spec.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import * as welshTranslations from '../../../../../../assets/i18n/cy.json';
+import * as englishTranslations from '../../../../../../assets/i18n/en.json';
+import { QuestionResult } from '@dvsa/mes-test-schema/categories/common';
+import { PopulateTestCategory } from '../../../../../../modules/tests/category/category.actions';
+import { PopulateCandidateDetails }
+  from '../../../../../../modules/tests/journal-data/cat-b/candidate/candidate.actions';
+import { candidateMock } from '../../../../../../modules/tests/__mocks__/tests.mock';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+
+describe('VehicleChecksCardComponent', () => {
```
Comment (MarkPatt89) [inline src/pages/debrief/cat-a-mod2/components/vehicle-checks-card/__tests__/vehicle-checks-card.spec.ts]: Should the describe mention `CatAMod2` for clarity of reading tests?

**Review by MarkPatt89 — APPROVED** [review]

**Pull request merged**